woocommerce/woocommerce-ios
# Product settings fetch

This PR updates the `SiteSettings` logic to now also download + cache the **product** settings values from the API. We were previously only fetching general settings, but now need the product settings so we can display the appropriate weight and length labels on the Product Details screen.

First off, please don't be scared by the diff size here 😄 The bulk of the changes (~1778 lines of 2,062) are the result of new test JSON files, XCode-generated updates to `WooCommerce.xcdatamodeld`, and boilerplate tests. Additionally, this is basically an all-or-nothing kind of change. Let me walk you through everything going on here:

### Networking
* A new `SiteSettingGroup` enum was added to the models
* `SiteSetting` has a new prop → `SiteSettingGroup` (used to classify a setting...right now it's just product and general)
* `SiteSettingsMapper` now requires a`SiteSettingGroup` during `init()`
* `SiteSettingsRemote` has a new `loadProductSettings` func
* Unit test additions (boilerplate)

### Storage
**There is a new CD model in this PR → v13**. The only change here is the addition of an optional `settingGroupKey` attribute on the existing `SiteSetting` entity. 

### Yosemite
* `SettingAction` and `SettingStore` now support fetching product _and_ general settings
* `StoresManager.synchronizeSettings()` has been lightly refactored to accommodate fetching multiple setting groups.
* Typical updates to `SiteSetting+ReadOnlyConvertible` and `SiteSetting+ReadOnlyType.swift`
* Unit test additions (boilerplate)

### WooCommerce
The only update in the app is to `CurrencySettings` in order to accommodate the new setting groups **and** also adding a `predicate` to the `resultsController` so we are fetching the correct store's settings (I think this was an undiscovered 🐛).

Fixes #807 

## Testing
* Review the updated code mentioned in the section above ☝️ 
* Make sure all of the unit tests are ✅ 
* Verify migrating from the `v12` to `v13` model is ✅  (this is a lightweight migration)
* Build and run the app:
    * Log out/in — verify the current store's currency settings are still fetched and applied correctly. 
    * Switch stores — verify the current store's currency settings are still fetched and applied correctly.
    * Using a SQLite browser verify the `SiteSetting` table contains `product` or `general` groups for each record.
